tsl/src/remote/dist_copy.c

```diff
@@ -813,9 +813,16 @@ remote_copy_process_and_send_data(RemoteCopyContext *context)
 	else
 		point = get_current_point_for_text_copy(context->data_context);
 
-	chunk = get_target_chunk(ht, point, &context->connection_state);
+	chunk = (Chunk *) get_target_chunk(ht, point, &context->connection_state);
 	connections = get_connections_for_chunk(context, chunk->fd.id, chunk->data_nodes, GetUserId());
 
+	/* for remote copy, we don't use chunk insert states on the AN.
+	 * so we need to explicitly set the chunk as unordered when copies
+	 * are directed to previously compressed chunks
+	 */
+	if (ts_chunk_is_compressed(chunk) && (!ts_chunk_is_unordered(chunk)))
+		ts_chunk_set_unordered(chunk);
```

Comment:
I think this should probably be part of `get_target_chunk`, which also fixes the const issue above.

Reply:
Won't that be setting it prematurely before the connections are created and available? I changed the status o ts_get_chunk to avoid the casting away of const.